src/stores/text_store.ts

```diff
@@ -5,148 +5,77 @@
-					title: this.textComponentTitle,
-					dimensions: {
-						width: 500,
-						height: 150
-					},
-					position: 'top',
-					cannotClose: true
-				}
-			}) as CreateComponentResponse;
-			this.textComponentID = tResult?.values?.id ?? -1;
-		}
-		await this.clearText();
-
-		// Take the focus away from the newly created text component
-		const tPluginID = await getComponentByTypeAndTitleOrName('game', kStoryQPluginName, kStoryQPluginName);
-		await codapInterface.sendRequest({
-			action: 'notify',
-			resource: `component[${tPluginID}]`,
-			values: {
-				request: 'select'
-			}
-		});
-	}
-
-	async clearText() {
-		const attributeName = pluralize(targetStore.targetAttributeName)
-		this.setTextSections([]);
-		await codapInterface.sendRequest({
-			action: 'update',
-			resource: `component[${this.textComponentID}]`,
-			values: {
-				text: {
-					"object": "value",
-					"document": {
-						"children": [
-							{
-								"type": "paragraph",
-								"children": [
-									{
-										"text": `This is where selected ${attributeName} appear.`
-									}
-								]
-							}
-						],
-						"objTypes": {
-							"paragraph": "block"
-						}
-					}
-				}
+  textComponentTitle: ITextPart[] = [];
+  textSections: ITextSection[] = [];
+  titleDataset: TitleDataset = "target";
+
+  constructor() {
+    makeAutoObservable(this, {}, { autoBind: true });
+  }
+
+  asJSON() {
+    return {
+      textComponentTitle: JSON.stringify(this.textComponentTitle)
+    };
+  }
+
+  fromJSON(json: ITextStoreJSON) {
+    if (json) {
+			try {
+				const title = JSON.parse(json.textComponentTitle);
+				this.textComponentTitle = title || [];
+			} catch (e) {
+				// The title used to be a simple string, so we have to accommodate that.
+				this.textComponentTitle = [{ text: json.textComponentTitle || "" }];
 			}
-		});
-	}
-
-	async closeTextComponent() {
-		await codapInterface.sendRequest({
-			action: 'delete',
-			resource: `component[${this.textComponentTitle}]`
-		});
-	}
+    }
+  }
+
+  setTextComponentTitle(title: ITextPart[]) {
+    this.textComponentTitle = title;
+  }
+
+  setTitleDataset(titleDataset: TitleDataset) {
+    this.titleDataset = titleDataset;
+  }
+
+  setTextSections(sections: ITextSection[]) {
+    this.textSections = sections;
+  }
+
+  getTextSectionId(textSection: ITextSection) {
+    return `section-${textSection.title?.actual}-${textSection.title?.predicted}`;
+  }
+
+  toggleTextSectionVisibility(textSection: ITextSection) {
+    textSection.hidden = !textSection.hidden;
+  }
+
+  updateTitle(datasetName: string, attributeName: string) {
+    if (datasetName && attributeName) {
+      this.textComponentTitle = [
+				{ text: `Selected ` },
+				{ text: pluralize(attributeName), classNames: ["highlighted"] },
+				{ text: " in " },
+				{ text: datasetName, classNames: ["highlighted"] }
+			];
+    } else {
+      this.textComponentTitle = [{ text: `Choose Data And Text To Begin` }];
+    }
```

Comment:
I haven't looked at this close enough. But it might be better to move this logic and text into a component so it isn't necessary to have a markdown/html like approach to generating multiline text in the stores.

Reply:
This was also a big improvement 👍